### ISSUE TITLE
False positive on Right operand cannot be mixed when using PHP_INT_MIN

### ISSUE THREAD

**Issue by simPod** [vimeo/psalm#4027]
```php
<?php

class X {
    private const DESCRIPTION = 'S' . PHP_INT_MIN;
}
```

![image](https://user-images.githubusercontent.com/327717/90607760-f6b31e80-e201-11ea-8d33-fa349508925f.png)

psalm v3.14.1 says that `PHP_INT_MIN` is mixed.   
Although I cannot reproduce it here https://psalm.dev/r/b442807487

Any idea what might be wrong?

**Comment by simPod** [comment]
Disappeared since v.3.14.1

**Comment by simPod** [comment]
It reappeared :/

**Comment by kgilden** [comment]
Noticing a similar problem in v3.16@d03e5ef0

**Comment by simPod** [comment]
You're right, clearing the cache helps.

**Comment by simPod** [comment]
Disappeared in v4, I'm curious whether it would return.

**Comment by simPod** [comment]
It's caused by `PHP_INT_MIN`. `PHP_INT_MAX` is fine.

**Comment by orklah** [comment]
Can you report on what environment is used to run psalm? PHP version, CI or not, phar or composer install...

**Comment by simPod** [comment]
Don't know anymore. Haven't seen it for a while.

**Comment by orklah** [comment]
Welp, then I'm gonna close this, no report since V4, which is now pretty old...
